velox/common/caching/SsdCache.cpp

```diff
@@ -40,23 +40,35 @@ SsdCache::SsdCache(
       numShards_(numShards),
       groupStats_(std::make_unique<FileGroupStats>()),
       executor_(executor) {
-  // Make sure the given path of Ssd files has the prefix for local file system.
-  // Local file system would be derived based on the prefix.
-  VELOX_CHECK(
-      filePrefix_.find("/") == 0,
-      "Ssd path '{}' does not start with '/' that points to local file system.",
-      filePrefix_);
-  filesystems::getFileSystem(filePrefix_, nullptr)
-      ->mkdir(std::filesystem::path(filePrefix).parent_path().string());
+  std::vector<std::string> prefixes;
+  std::istringstream iss(filePrefix_);
+  std::string prefix;
+  while (std::getline(iss, prefix, ',')) {
+    // Make sure the given path of Ssd files has the prefix for local file
+    // system. Local file system would be derived based on the prefix.
+    VELOX_CHECK(
+        prefix.find("/") == 0,
+        "Ssd path '{}' does not start with '/' that points to local file system.",
+        prefix);
+    filesystems::getFileSystem(prefix, nullptr)
+        ->mkdir(std::filesystem::path(prefix).parent_path().string());
+    prefixes.push_back(prefix);
+  }
+  std::random_device rd;
+  std::default_random_engine engine(rd());
 
   files_.reserve(numShards_);
   // Cache size must be a multiple of this so that each shard has the same max
   // size.
   uint64_t sizeQuantum = numShards_ * SsdFile::kRegionSize;
   int32_t fileMaxRegions = bits::roundUp(maxBytes, sizeQuantum) / sizeQuantum;
   for (auto i = 0; i < numShards_; ++i) {
+    // Shuffle the prefixes to avoid hotspotting.
+    std::shuffle(prefixes.begin(), prefixes.end(), engine);
```

Comment:
For simplicity, can we just do one-time shuffle to generate a random sequence? Then, just select the prefix in round-robin way. Not sure whether they have any essential difference.

Reply:
Hm, the main purpose of multiple `std::shuffle` is to makesure that different shards can be allocated randomly, especially when there are multiple velox processes on the same machine (such as multiple gluten-based executors), so that the disks can be distributed as evenly as possible and hot spots can be reduced.